editors/sc-ide/core/sc_process.cpp

```diff
@@ -39,12 +39,18 @@ namespace ScIDE {
 
 ScProcess::ScProcess( Settings::Manager * settings, QObject * parent ):
     QProcess( parent ),
-    mIpcServer( new QLocalServer(this) ),
-    mIpcSocket(NULL),
+    mIpcServer( new QTcpServer(this) ),
     mIpcServerName("SCIde_" + QString::number(QCoreApplication::applicationPid())),
     mTerminationRequested(false),
     mCompiled(false)
 {
+
+	mIpcServer->listen(QHostAddress(QHostAddress::LocalHost), ScIpcChannel::Port);
+
+	// this is initialized when connecting.
+	mIpcChannel = nullptr;
+	mIpcSocket = nullptr;
```

Comment:
please initialise the members either in the class (we are finally using c++11) or as in the code you've just removed.